executor/union_scan.go

```diff
@@ -28,12 +29,17 @@ import (
 // DirtyDB stores uncommitted write operations for a transaction.
 // It is stored and retrieved by context.Value and context.SetValue method.
 type DirtyDB struct {
+	sync.Mutex
+
 	// tables is a map whose key is tableID.
 	tables map[int64]*DirtyTable
 }
 
 // GetDirtyTable gets the DirtyTable by id from the DirtyDB.
 func (udb *DirtyDB) GetDirtyTable(tid int64) *DirtyTable {
+	// The index join access the tables map paralelly.
+	// But the map throws panic in this case. So it's locked.
```

Comment:
Could you also update the comment in table reader case of `buildUnionScanFromReader`? It is outdated now.

Reply:
Done.